kebechet_administrator.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+# workflow-helpers Kebechet Administrator
+# Copyright(C) 2020 Sai Sankar Gochhayat
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Kebechet Administrator.
+
+This script run in a workflow task to take an incoming message and decides which repositories Kebechet
+needs to be run on and store the necessary messages to be sent.
+"""
+
+import logging
+import json
+import semver
+from typing import Dict
+from thoth.storages import GraphDatabase
+from thoth.workflow_helpers.configuration import Configuration
+from thoth.messaging import __all__ as all_messages
+from thoth.workflow_helpers import __service_version__
+
+__COMPONENT_NAME__ = "Kebechet Administrator"
+
+_LOGGER = logging.getLogger("thoth.run_kebechet_administrator")
+_LOGGER.info("Thoth workflow-helpers task: run_kebechet_administrator v%s", __service_version__)
+
+# Estabilish connection to Database.
+GRAPH = GraphDatabase()
+GRAPH.connect()
+
+_URL_PREFIX = "https://github.com/"
+
+output_messages = []  # Messages to be sent by producer.
+
+
+def _handle_solved_message(Configuration):  # noqa: N803
+    """Handle all the messages for which Kebechet needs to run on if the sovler type matches the os type."""
+    solver_string = Configuration.get("THOTH_SOLVER_NAME")  # ex - solver-fedora-31-py38
+    if not solver_string:
+        raise ValueError(
+            f"SolverMessageType has been provided to the MESSAGE_TYPE env variable. \
+            but solver name is missing."
+        )
+    _, os_name, os_version, python_version = solver_string.rsplit(sep="-", maxsplit=3)
+    python_version = ".".join([i for i in python_version if i.isdigit()])  # generates '3.8' from 'py39'
```

Comment:
Will this work for Python 3.10?

Reply:
Actually no it will fail if the input string is 'py310' any suggestions?

Reply:
Please use method from common so we can anytime in the future adjust this logic:

https://github.com/thoth-station/common/blob/11215967883ffbefc6c14c5a0ac2b871114ffb09/thoth/common/openshift.py#L203

Reply:
Thanks

Reply:
Now using parse_solver_method from common.